### PR TITLE
Create stub template, change POLICIES.md/CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by lemoncurry**
---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by lemoncurry** [comment]
Before adding the missing stub files, we should agree on the template.

**Comment by lemoncurry** [comment]
The installation of jdk8 failed, so I had to update the travis config file and close/open the PR to trigger the builds.

See https://github.com/exercism/java/pull/1724.

**Comment by Smarticles101** [comment]
(@lemoncurry as a maintainer you should be able to restart the build from [here](https://travis-ci.org/exercism/java/pull_requests) if you sign in with github)

File: POLICIES.md [hunk 1]
```diff
@@ -36,9 +36,9 @@ References: [[1](https://github.com/exercism/java/issues/177#issuecomment-261291
 > - Exercises of difficulty 4 or lower: provide stubs for all required constructors and methods. This means that you need to provide stubs for those constructors or methods that are necessary to pass all tests. E.g. stubs for private methods may be skipped.
 Stubs should include the following body:
     `throw new UnsupportedOperationException("Delete this statement and write your own implementation.");`
-> - Exercises of difficulty 5 or higher: provide no stubs by default, but either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.
+> - Exercises of difficulty 5 or higher: provide a starter file only (StubTemplate.java) and rename it, but either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.
```
Comment (Smarticles101) [inline POLICIES.md]:
```suggestion
> - Exercises of difficulty 5 or higher: provide a starter file without stubs (StubTemplate.java) and rename it, and either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.
```

**Comment by lemoncurry** [comment]
> (@lemoncurry as a maintainer you should be able to restart the build from [here](https://travis-ci.org/exercism/java/pull_requests) if you sign in with github)

Thank you!

File: POLICIES.md [hunk 2]
```diff
@@ -36,9 +36,9 @@ References: [[1](https://github.com/exercism/java/issues/177#issuecomment-261291
 > - Exercises of difficulty 4 or lower: provide stubs for all required constructors and methods. This means that you need to provide stubs for those constructors or methods that are necessary to pass all tests. E.g. stubs for private methods may be skipped.
 Stubs should include the following body:
     `throw new UnsupportedOperationException("Delete this statement and write your own implementation.");`
-> - Exercises of difficulty 5 or higher: provide no stubs by default, but either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.
+> - Exercises of difficulty 5 or higher: provide a starter file without stubs (StubTemplate.java) and rename it, and either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.
```
Comment (FridaTveit) [inline POLICIES.md]: Might be useful to make it more clear where they should copy this file from? 🙂

Comment (lemoncurry) [reply POLICIES.md]: Thank you @FridaTveit , good point. I will add this information.

File: POLICIES.md [hunk 3]
```diff
@@ -36,7 +36,7 @@ References: [[1](https://github.com/exercism/java/issues/177#issuecomment-261291
 > - Exercises of difficulty 4 or lower: provide stubs for all required constructors and methods. This means that you need to provide stubs for those constructors or methods that are necessary to pass all tests. E.g. stubs for private methods may be skipped.
 Stubs should include the following body:
     `throw new UnsupportedOperationException("Delete this statement and write your own implementation.");`
-> - Exercises of difficulty 5 or higher: provide a starter file without stubs (StubTemplate.java) and rename it, and either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.
+> - Exercises of difficulty 5 or higher: provide a starter file without stubs (StubTemplate.java file provided [here](https://github.com/exercism/java/tree/master/_template/src/main/java)) and rename it, and either (1) add hints to the hints.md file (which gets merged into the README.md for the exercise) or (2) provide stubs as above for exercises that demand complicated method signatures.
```
Comment (FridaTveit) [inline POLICIES.md]: Sorry, I think this sentence is still a tiny bit confusing. Reading it I think it's not quite clear if you're meant to copy the provided file or create your own. Something like:
```
Exercises of difficulty 5 or higher: copy the StubTemplate.java file (provided [here](...)) and rename it, and either ...
```
might be more clear? 🙂 And maybe a sentence explaining what we'd like it to be renamed to could be useful? E.g. `... and rename it to fit the exercise. For example, for the exercise linked-list the file could be named LinkedList.java`. That's probably obvious, but maybe better to add too much information than too little? 🙂

Comment (lemoncurry) [reply POLICIES.md]: Yes, that's true, it's better to add more information. I added your suggestions.

**Pull request merged**